### PR TITLE
Add module's shared functions even when its execute() raised an excep…

### PR DESCRIPTION
**Pull request by happz**
…tion

We want to give a module chance to provide at least some level of its
functionality even when it killed the pipeline. By our observations,
quite often following modules are interested in data the culprit could
provide, even when they migh be incomplete.

### CODE REVIEW

File: gluetool/glue.py [hunk 1]
```diff
@@ -1514,8 +1514,24 @@ def _sanity(module):
...
+            # but should there be an exception under ``try`` *and* should there be another
+            # one under ``finally``, the first one would be lost, replaced by the later.
+            # And we cannot guarantee exception-less ``add_shared``, it may have been replaced
+            # by module's developer. Therefore resorting to being more verbose.
+            try:
+                module.execute()
+
+            # pylint: disable=broad-except,unused-variable
+            except Exception as exc:  # noqa
+                # In case ``add_shared`` crashes, the exception is not lost since it's already
```
Comment (thrix) [inline gluetool/glue.py]: maybe change the wording please. ```the exception``` -> ```the previous execption```

Comment (thrix) [reply gluetool/glue.py]: or even ```the previous exception from execute```

File: gluetool/tests/test_run_modules.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    assert glue.has_shared('foo')
+
+    assert [
+        member[1] for member in inspect.getmembers(glue.get_shared('foo'), inspect.isclass) if member[0] == 'im_class'
+    ].index(BrokenModule) == 0
+
+
+"""
+def test_core_add_shared(glue):
```
Comment (thrix) [inline gluetool/tests/test_run_modules.py]: why is this here?

**Comment by thrix** [comment]
otherwise LGTM

**Comment by happz** [comment]
@thrix "otherwise"? :) I see no comments (and no "Approved" as well :)

**Comment by thrix** [comment]
@happz hmm, I do not see addressed my two small comments? ^

**Comment by happz** [comment]
@thrix Nope, I don't see them. Anyway, I changed the comment, based on the image you sent. Ready for review.

**Review by thrix — CHANGES_REQUESTED** [review]

File: gluetool/tests/test_run_modules.py (hunk 2 above)
Comment (happz) [reply gluetool/tests/test_run_modules.py]: Ah, right. I copy & pasted some test cases to refactor them, and didn't remove the rest. Removed.

File: gluetool/glue.py (hunk 1 above)
Comment (happz) [reply gluetool/glue.py]: Should be fixed.

**Review by thrix — APPROVED** [review]
LGTM now

**Comment by happz** [comment]
Given there are no other reviewers available at this moment, going to merge with just a single review.

**Pull request merged**